src/lib/y2network/connection_config/base.rb

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/interface"
+
+module Y2Network
+  module ConnectionConfig
+    # This class is reponsible of a connection configuration
+    class Base
+      # @return [Interface, nil]
```

Comment:
what means if Interface is nil? is it valid config or not?

Reply:
It is open to discuss. You could define a connection configuration not attached to any specific interface, so the first resource / device available will apply the configuration

Reply:
ok, so please document it what it stands for. As at first look it looks strange, but if nil works as * wildcard, it is fine for me.

Reply:
We will not apply it yet, I mean, by now, with wicked as the target, all the connections will have an interface associated.